docs/low-code-spark/gems/join-split/join.md

```diff
@@ -4,20 +4,211 @@ title: Join
+|:----------------------------------|:----------------------------------------------------------------------------|:---------------------------------------|
+| Dataframe 1                       | First input dataframe                                                       | True                                   |
+| Dataframe 2                       | Second input dataframe                                                      | True                                   |
+| Dataframe N                       | Nth input dataframe                                                         | False                                  |
+| Join Condition ( Conditions tab ) | The join condition specifies how the rows will be combined.                 | True                                   |
+| Type ( Conditions tab )           | The type of JOIN (  Inner, Full Outer, Left , Right , Left Semi, Left Anti  | True                                   |
+| Where Clause ( Conditions tab )   | Filter applied after the Join operation                                     | False                                  |
+| Target column ( Expressions )     | Output column name                                                          | False                                  |
+| Expression   ( Expressions )      | Expression to compute target column                                         | Required if a Target column is present |
+| Type ( Hints )                    | The type of Hint ( Broadcast, Merge, Shuffle Hash, Shuffle Replicate NL     | False                                  |
+
+### Add a new input
+1. Click on the plus icon to add a new input.
+2. Then add your condition expression for the newly added input.
+   ![Example usage of Aggregate - Aggregation with Grouping](./img/add_new_input.png)
+
+### Examples
+
+#### Example 1 - Join with three datasets
+![Example usage of Aggregate - Aggregation with Grouping](./img/join_without_hints.png)
+````mdx-code-block
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs>
+<TabItem value="py" label="Python">
+
+```py
+def total_orders(spark: SparkSession, in0: DataFrame) -> DataFrame:
+        return in0\
+        .alias("in0")\
+        .join(in1.alias("in1"), (col("in0.customer_id") == col("in1.customer_id")), "inner")\
+        .join(in2.alias("in2"), (col("in1.customer_id") == col("in2.customer_id")), "inner")
+```
+
+</TabItem>
+
+<TabItem value="scala" label="Scala">
+
+```scala
+object total_orders {
+   def apply(
+    spark: SparkSession,
+    in0:   DataFrame,
+    in1:   DataFrame,
+    in2:   DataFrame
+  ): DataFrame =
+    in0
+      .as("in0")
+      .join(in1.as("in1"), col("in0.customer_id") === col("in1.customer_id"), "inner")
+      .join(in2.as("in2"), col("in1.customer_id") === col("in2.customer_id"), "inner")
+}
+```
+
+</TabItem>
+</Tabs>
+
+````
+
+#### Example 2 - Join with Hints
+
+![Example usage of Aggregate - Aggregation with Grouping](./img/join_with_hints.png)
```

Comment:
Can each of these skew hints be configured in Prophecy?
https://docs.databricks.com/delta/join-performance/skew-join.html
Perhaps we just want to reference the databricks doc for use as needed?

Reply:
If we offer broadcast, merge, shuffle hash, shuffle NL hints, then we should provide rationale for how to chose which type of hint. This can be a link to an external doc.

Reply:
Good call, let me add it

Reply:
In addition to the above two doc links, I suggest adding a link to educate users how to create a WHERE clause such that the filter would be pushed down, i.e. only read the data included in the filter THEN do the join.

Reply:
@atbida With Databricks Runtime 7.3 and above, skew join hints are not required. Skew is automatically taken care of if adaptive query execution (AQE) and spark.sql.adaptive.skewJoin.enabled are both enabled. See [Adaptive query execution](https://docs.databricks.com/spark/latest/spark-sql/aqe.html).